tests/extensions/pdf/test_exporter.py

```diff
@@ -0,0 +1,93 @@
+import os
+import shutil
+
+import pytest
+
+from mfr.extensions.pdf import settings
+from mfr.extensions.pdf import exceptions
+from mfr.extensions.pdf import PdfExporter
+
+BASE = os.path.dirname(os.path.abspath(__file__))
+
+
+@pytest.fixture
+def directory(tmpdir):
+    return str(tmpdir)
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup_filesystem(directory):
+    shutil.rmtree(directory)
+    os.makedirs(directory, exist_ok=True)
+
+
+class TestPdfExporter:
+    '''
+    Opening and veryifying pdfs with report lab is a paid feature
+    so these tests are mostly just to see if the file gets created
+    in the correct place
+    '''
+
+    def test_single_page_tiff(self, directory):
+        source_file_path = os.path.join(BASE, 'files', 'test.tif')
+        output_file_path = os.path.join(directory, 'test.{}'.format(settings.EXPORT_TYPE))
+        format = '{}.{}'.format(settings.EXPORT_MAXIMUM_SIZE, settings.EXPORT_TYPE)
+        exporter = PdfExporter(source_file_path=source_file_path,
+                            output_file_path=output_file_path, format=format)
+
+        assert not os.path.exists(output_file_path)
+        assert os.path.exists(directory)
+
+        exporter.export()
+        assert os.path.exists(output_file_path)
+
+    def test_multi_page_tiff(self, directory):
+        source_file_path = os.path.join(BASE, 'files', 'test_multipage.tif')
+        output_file_path = os.path.join(directory, 'test.{}'.format(settings.EXPORT_TYPE))
+        format = '{}.{}'.format(settings.EXPORT_MAXIMUM_SIZE, settings.EXPORT_TYPE)
+        exporter = PdfExporter(source_file_path=source_file_path,
+                            output_file_path=output_file_path, format=format)
+
+        assert not os.path.exists(output_file_path)
+        assert os.path.exists(directory)
+
+        exporter.export()
+        assert os.path.exists(output_file_path)
+
+    def test_ratio_tiff(self, directory):
+        source_file_path = os.path.join(BASE, 'files', 'test_ratio.tif')
+        output_file_path = os.path.join(directory, 'test.{}'.format(settings.EXPORT_TYPE))
+        format = '{}.{}'.format(settings.EXPORT_MAXIMUM_SIZE, settings.EXPORT_TYPE)
+        exporter = PdfExporter(source_file_path=source_file_path,
+                            output_file_path=output_file_path, format=format)
+
+        assert not os.path.exists(output_file_path)
+        assert os.path.exists(directory)
+
+        exporter.export()
+        assert os.path.exists(output_file_path)
```

Comment:
It looks like the successful tests and the failing tests could be parameterized, since we don't have the ability to do deep examination of the pdfs.

Reply:
What do you mean by this?

Reply:
The three passing tests are all very similar, they just take slightly different arguments.  You could parameterize them a la [this WB s3 test](https://github.com/CenterForOpenScience/waterbutler/blob/develop/tests/providers/s3/test_provider.py#L119).  That way if you update one later, the others get updated, too.

Reply:
This is great,
completed.